### DISCUSSION TITLE
edge function timeout, a definitive answer?

### DISCUSSION THREAD

**Discussion by mkandan** [supabase/supabase#13837] [Questions] [answered]
is there a definitive timeout for edge functions? i haven't been able to find much documentation except for [this reddit post](https://www.reddit.com/r/Supabase/comments/1197v5u/edge_function_time_limit/)

I'm triggering (via HTTP request) a Python 3.9 runtime Google Cloud Function from an edge function, but it can run anywhere from 5s - 4 min (probably more) depending on the user's input.

**Comment by mlanza** [comment]
I too am confused by what's stated.

https://deno.com/deploy/docs/pricing-and-limits

This page mentions 10ms (or 50s) per request, but in my experience a function can run for several seconds, the post says "minutes."  So not sure how to make sense of that.  My function is pure and receives all it's data directly.  It's run for 7-ish seconds before being stopped by free-tier limits.

Possibly its about [the difference between CPU time and wall clock time](https://stackoverflow.com/questions/68720436/what-is-cpu-time-and-wall-time-in-the-context-of-cloudflare-worker-request).  What confuses me is that my function is pure.  So I don't know why there'd be any disparity between CPU time and wall clock time (e.g. what's it waiting on?).

**Comment by mansueli** [comment, marked as answer]
It is currently 100ms of compute time (CPU time), but we don't have a determined set for wall time.

Essentially the timeout in those cases comes from the service where we deployed the functions. We are working to improve this experience with more customized options in the future. 

The only way to tweak this at the moment is by self-hosting the [Edge Functions](https://github.com/supabase/edge-runtime/).

**Reply by JS-GitRepo** [reply]
Firebase / Google Cloud Functions seem to only measure in wall time from all resources I can find for their timeouts. Having a hard time comparing / knowing if my existing Firebase Cloud Functions (that have a wall time of 10-15 seconds) would work with Supabase, but I want to make my next product there 🤔

**Reply by JulienLecoq** [reply]
Any news on that? I would also be very interested to be able to increase the CPU time limit. I have an Edge function that was working before which does generate a PDF report, but now the reports that I need to generate are bigger thus it often not works anymore.

**Reply by reilly1993** [reply]
How do you tweak it when self-hosting? Is there a config file somewhere?

**Comment by mlanza** [comment]
It would be nice if that time could be controlled by devs if only so that they can easily identify whether increasing it solves a problem.  Obviously, if it does, they'll have to optimize the function to bring the time down, but it does provide a way to slide by with an inefficient computation for a time, even if at added cost.

**Comment by pedromassango** [comment]
My function fails when uploading files with 9+ MBs. took me some time to figure that this was the problem as the crash error message was not explanatory.

**Comment by pradeepb28** [comment]
is 100ms still actual? 
I see my edge functions were able to run more than 3 seconds (all the time with no issues)

**Reply by JulienLecoq** [reply]
Yes, it's still 100 ms for CPU time. For edge function run time, I don't know the limit.

**Reply by burakcan** [reply]
Is there a wall time limit? I have a function which uses around 8ms of CPU but takes well over 2 minutes of wall time. Would it cause problems?

**Comment by mansueli** [comment]
We documented the limits on this page:

https://supabase.com/docs/guides/functions/debugging#limitations


Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by rajivsinclair** [reply]
If I am self-hosting Supabase (with Supabase Edge Functions Runtime), then can the max execution time be configured in `config.toml`? [Perplexity/GPT-4](https://www.perplexity.ai/search/How-do-I-4QcjNCY0RlGzOWpvMhP.ig#3) seems to think so, but I can’t find any officially documented source that confirms this.

**Reply by rolandpeelen** [reply]
Updated link here: https://supabase.com/docs/guides/functions/limits - Old one gives a 404

**Reply by pedromassango** [reply]
> Updated link here: https://supabase.com/docs/guides/functions/limits - Old one gives a 404

Please, also make it clear the file upload limit with Functions.
